### DISCUSSION TITLE
Looping, transitionConditionally and onEntry

### DISCUSSION THREAD

**Discussion by zmullett** [nsk90/kstatemachine#58] [Q&A] [answered]
I have a state that should be looped in until a terminating condition is met. I would like that the `onEntry` block is called after each recurrence.

```kotlin
repeatableState {
  onEntry { /* do some important stuff repeatedly */ }

  transitionConditionally<SomeEvent> {
    direction = { targetState(if (anotherRound()) repeatableState else idleState) }
  }
}
```

It seems that the machine doesn't re-enter the state to call `onEntry`, which one might assume is correct behavior since it's not really _entering_ again. However I don't see any alternatives in the source.

Is there a way to express this machine such that a block is called when first arriving at the state, then repeatedly if looping?

Thanks!

**Comment by nsk90** [comment, marked as answer]
Hi, I am on vacation now, so cannot go deep into your question. But try changing transitionType argument of your transition to "external". In such case machine should reenter your state if it is already entered. If it does not help I will come back to your question later.) Happy new year!

**Reply by zmullett** [reply]
Confirmed this works. Happy new year! 🥳
